test/elixir_bench_web/schema/schema_test.exs

```diff
@@ -79,30 +81,403 @@ defmodule ElixirBenchWeb.SchemaTest do
        }
 
       {:ok, resp} = context.conn
-        |> post("/api", AbsintheHelpers.query_skeleton(query))
-        |> Map.get(:resp_body)
-        |> Antidote.decode
+        |> execute_query(query)
+        |> decode_response_body
 
       assert ^json_data = resp
     end
 
-    test "not break when repo not found", context do
+    test "return not found message when repo does not exist", context do
       query = """
         repo (slug: "Idont/exist") {
           name
         }
       """
 
       {:ok, resp} = context.conn
-        |> post("/api", AbsintheHelpers.query_skeleton(query))
+        |> execute_query(query)
+        |> decode_response_body
+
+      error_message = Enum.at(resp["errors"], 0)
+
+      assert %{"message" => "not_found"} = error_message
+    end
+
+    test "return required fields message when missing fields", context do
+      required_fields = ["slug"]
+
+      query = """
+        repo {
+          name
+        }
+      """
+
+      resp_body = context.conn
+        |> execute_query(query)
         |> Map.get(:resp_body)
-        |> Antidote.decode
+
+      Enum.each(required_fields, fn field ->
+        assert(
+          Regex.match?(~r/errors.*#{field}.*found null/, resp_body),
+          "Field '#{field}' not found in #{resp_body}"
+        )
+      end)
+    end
+
+    test "return slug error message when invalid slug is given", context do
+      query = """
+        repo (slug: "invalidSlug") {
+          name
+        }
+      """
+
+      {:ok, resp} = context.conn
+        |> execute_query(query)
+        |> decode_response_body
+
+      error_message = Enum.at(resp["errors"], 0)
+
+      assert %{"message" => "invalid_slug"} = error_message
+    end
+  end
+
+  describe "benchmark query" do
+    test "fetch benchmark by name and repo slug", context do
+      benchmark = insert(:benchmark, %{name: "insert_mysql/insert_plain"})
+      repo = benchmark.repo
+
+      query = """
+        benchmark (
+          repo_slug: "#{repo.owner}/#{repo.name}",
```

Comment:
Seems to me like we could make use of a `slug` function in the repo to create this as we seem to repeat this logic in quite some places.

Reply:
I created an issue for that. #13